service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/auth/IotHubServiceSasToken.java

```diff
@@ -117,10 +134,19 @@ private String buildToken()
     private long buildExpiresOn()
     {
         long expiresOnDate = System.currentTimeMillis();
-        expiresOnDate += TOKEN_VALID_SECS * 1000;
+        expiresOnDate += this.tokenLifespanSeconds * 1000;
         return expiresOnDate / 1000;
     }
 
+    /**
+     * @return The number of seconds that this token is valid for. Not to be confused with how many seconds the SAS token
+     * is still valid for at the time of calling this method.
```

Comment:
is there a way to get "how many seconds the SAS token is still valid for at the time of calling this method." directly, or is it mathematical calculation? worth adding the equation here (even though it might be simple subtraction)

Reply:
if we are being _that_ thorough!

Reply:
Thankfully we do have a method for that already:

```java
    public final long getExpiryTimeMillis();
```

Reply:
No math required, other than converting milliseconds to seconds if necessary